### PR TITLE
Annotate System.Net.Mail for nullable reference types

### PR DESCRIPTION
**Pull request by stephentoub**
Contributes to #2339 
cc: @buyaa-n, @eiriktsarpalis, @safern, @davidsh

### CODE REVIEW

File: src/libraries/Common/src/System/Net/NTAuthentication.Common.cs [hunk 1]
```diff
@@ -16,7 +17,7 @@ internal partial class NTAuthentication
 
         private SafeFreeCredentials? _credentialsHandle;
         private SafeDeleteContext? _securityContext;
-        private string _spn = null!;
+        private string? _spn = null!;
```
Comment (buyaa-n) [inline src/libraries/Common/src/System/Net/NTAuthentication.Common.cs]: NIT: could you remove `!`

File: src/libraries/System.Net.Mail/src/System/Net/Mail/Attachment.cs [hunk 2]
```diff
@@ -220,15 +220,15 @@ public Stream ContentStream
                     throw new ObjectDisposedException(GetType().FullName);
                 }
 
-                return _part.Stream;
+                return _part.Stream!;
             }
         }
 
         public string ContentId
```
Comment (buyaa-n) [inline src/libraries/System.Net.Mail/src/System/Net/Mail/Attachment.cs]: Might not intended but it is allowing null

File: src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs [hunk 3]
```diff
@@ -26,22 +26,20 @@ internal partial class SmtpConnection
...
-        internal SmtpTransport _parent;
-        private readonly SmtpClient _client;
-        private NetworkStream _networkStream;
-        internal TcpClient _tcpClient;
-        internal string _host = null;
+        private readonly EventHandler? _onCloseHandler;
+        internal SmtpTransport? _parent;
+        private readonly SmtpClient? _client;
+        private NetworkStream? _networkStream;
+        internal TcpClient? _tcpClient;
```
Comment (buyaa-n) [inline src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs]: NIT: I might miss something but `_client` and `_tcpClient` doesn't seem to be nullable

**Review by buyaa-n — APPROVED** [review]
Looks good, thanks! Left few NIT comment.

File: src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs (hunk 3 above)
Comment (stephentoub) [reply src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs]: SmtpConnection is partial, and it has this ctor in another file:
https://github.com/dotnet/runtime/blob/5e6711dbfb1b696e1fb7c53f9d84d33c60dc87be/src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.Auth.cs#L35-L38

**Comment by stephentoub** [comment]
Thanks for the review, @buyaa-n.

**Pull request merged**